src/database/engine/page.c

```diff
@@ -510,6 +553,43 @@ void pgd_append_point(PGD *pg,
 
             break;
         }
+        case PAGE_CONSTANT_METRICS: {
+            storage_number t = pack_storage_number(n, flags);
+
+            if (pg->used == 0)
+            {
+                pg->constant.v = t;
+                pg->used = 1;
+            }
+            else if (pg->constant.v == t)
+            {
+                pg->used++;
+            }
+            else
+            {
+                storage_number v = pg->constant.v;
+
+                global_statistic_constant_page_rm();
+                global_statistic_raw_page_new();
+
+                pg->type = PAGE_RAW_METRICS;
+                uint32_t size = pg->slots * page_type_size[pg->type];
+                pg->raw.size = size;
+                pg->raw.data = pgd_data_aral_alloc(size);
+
+                storage_number *data = (storage_number *) pg->raw.data;
+                for (uint32_t i = 0; i != pg->used; i++)
```

Comment:
In most cases, this will happen during the first couple points collected in a page, ie. there should not be any noticeable CPU overhead.